lua/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ target_include_directories(libopensnlua
     $<INSTALL_INTERFACE:include/opensn>
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
     $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${PROJECT_SOURCE_DIR}
```

Comment:
The reason for this change is that this added `<repo_root>/lua` into include path, so people could omit the `lua` prefix in the `#include`s. After this change they have to use the path from the repo_root, i.e. include the `lua` subdirectory.